homeassistant/components/zha/api.py

```diff
@@ -987,12 +987,16 @@ async def async_binding_operation(
             )
         )
     res = await asyncio.gather(*(t[0] for t in bind_tasks), return_exceptions=True)
+    errors = False
     for outcome, log_msg in zip(res, bind_tasks):
         if isinstance(outcome, Exception):
             fmt = f"{log_msg[1]} failed: %s"
+            errors = True
         else:
             fmt = f"{log_msg[1]} completed: %s"
         zdo.debug(fmt, *(log_msg[2] + (outcome,)))
+    if errors:
+        raise Exception("Binding operation failed")
```

Comment:
Can we raise a more specific `Exception` subclass and possibly include `errors` within the message string?

Reply:
Should it be a home assistant or zigpy based exception?

Reply:
I can raise whatever exception we wanna raise. I wasn’t sure how much to do here. For things like battery devices if they aren’t awake I want the feedback in the UI that the command failed but I don’t want issues on GitHub for this. These aren't things we can fix for the most part. It's reading attributes, writing, binding and issuing cluster commands. Let me know how you want to handle this and I'll adjust the PR accordingly.

Reply:
https://developers.home-assistant.io/docs/integration_quality_scale_index?_highlight=quality#silver-
Raise ValueError on invalid user input and raise HomeAssistantError for other failures such as a problem communicating with a device.

Reply:
Since there's no user input on binding, this means binding errors from device/network should result in HomeAssistant errors.

Reply:
I will handle these in different PR's. I am redesigning these controls. The first PR is here: #79908